stubs/colorful/colorful/core.pyi

```diff
@@ -0,0 +1,100 @@
+_PaletteType: TypeAlias = dict[_str, _str | tuple[int, int, int]]
+_SanitizedPaletteType: TypeAlias = dict[_str, tuple[int, int, int]]
+_StyleType: TypeAlias = tuple[_str, _str]
+
+DEFAULT_RGB_TXT_PATH: _str
+COLOR_PALETTE: dict[_str, _str]
+COLORNAMES_COLORS_PATH: _str
+
+class ColorfulError(Exception): ...
+class ColorfulAttributeError(AttributeError, ColorfulError): ...
+
+def translate_rgb_to_ansi_code(red: int, green: int, blue: int, offset: int, colormode: _ColorModeType) -> _str: ...
+def translate_colorname_to_ansi_code(
+    colorname: _str, offset: int, colormode: _ColorModeType, colorpalette: _PaletteType
+) -> _str: ...
+def resolve_modifier_to_ansi_code(modifiername: _str, colormode: _ColorModeType) -> _str: ...
+def translate_style(style: _str, colormode: _ColorModeType, colorpalette: _PaletteType) -> _str: ...
+def style_string(string: _str, ansi_style: _StyleType, colormode: _ColorModeType, nested: bool = False) -> _str: ...
+
+class ColorfulString:
+    orig_string: _str
+    styled_string: _str
+    colorful_ctx: Colorful
+    def __init__(self, orig_string: _str, styled_string: _str, colorful_ctx: Colorful) -> None: ...
+    def __len__(self) -> int: ...
+    def __iter__(self) -> Iterator[_str]: ...
+    def __add__(self, other: _str | ColorfulString) -> Self: ...
+    def __iadd__(self, other: _str | ColorfulString) -> Self: ...
+    def __radd__(self, other: _str | ColorfulString) -> Self: ...
+    def __mul__(self, other: _str) -> Self: ...
+    def __format__(self, format_spec: _str) -> _str: ...
+    def __getattr__(self, name: _str) -> Any: ...
+
+class Colorful:
+    NO_COLORS: int
+    ANSI_8_COLORS: int
+    ANSI_16_COLORS: int
+    ANSI_256_COLORS: int
+    TRUE_COLORS: int
+    COLORNAMES_COLORS = COLORNAMES_COLORS_PATH
+    close_fg_color: _str
+    close_bg_color: _str
+    no_bold: _str
+    no_dimmed: _str
+    no_italic: _str
+    no_underlined: _str
+    no_blinkslow: _str
+    no_blinkrapid: _str
+    no_inversed: _str
+    no_concealed: _str
+    no_struckthrough: _str
+    colormode: _ColorModeType
+    def __init__(self, colormode: _ColorModeType | None = None, colorpalette: _str | _PaletteType | None = None) -> None: ...
+    @property
+    def colorpalette(self) -> _SanitizedPaletteType | None: ...
+    @colorpalette.setter
+    def colorpalette(self, colorpalette: _str | _PaletteType) -> None: ...
+    def setup(
+        self,
+        colormode: _ColorModeType | None = None,
+        colorpalette: _str | _PaletteType | None = None,
+        extend_colors: bool = False,
+    ) -> None: ...
+    def disable(self) -> None: ...
+    def use_8_ansi_colors(self) -> None: ...
+    def use_16_ansi_colors(self) -> None: ...
+    def use_256_ansi_colors(self) -> None: ...
+    def use_true_colors(self) -> None: ...
+    def use_palette(self, colorpalette: _str | _PaletteType) -> None: ...
+    def update_palette(self, colorpalette: _str | _PaletteType) -> None: ...
```

Comment:
Same here.

Reply:
here im a bit confused i think `_PaletteType` itself is updated but these should be `SupportsItems[_str, | _PaletteType]` ?

its not really clear right now what the issue is here :confused:

Reply:
@srittau did you got the chance to look at this? I'm happily update that to be correct but right now i don't really understand how to do that